pkg/controller/managedcluster/managed_cluster_controller.go

```diff
@@ -125,9 +155,25 @@ func clusterType(cluster *mcv1.ManagedCluster) string {
 	return "Unknown"
 }
 
-func newKlusterletAddonConfig(clusterType string, name string) *kacv1.KlusterletAddonConfig {
-	switch clusterType {
-	case clusterTypeClusterClaim:
+func newKlusterletAddonConfig(clusterType string, name string, hostedAddOnEnabled bool) *kacv1.KlusterletAddonConfig {
+	switch {
+	case hostedAddOnEnabled:
+		return &kacv1.KlusterletAddonConfig{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: name,
+				Name:      name,
+			},
+			Spec: kacv1.KlusterletAddonConfigSpec{
+				ClusterName:                name,
+				ClusterNamespace:           name,
+				ApplicationManagerConfig:   kacv1.KlusterletAddonAgentConfigSpec{Enabled: false},
+				CertPolicyControllerConfig: kacv1.KlusterletAddonAgentConfigSpec{Enabled: false},
+				IAMPolicyControllerConfig:  kacv1.KlusterletAddonAgentConfigSpec{Enabled: false},
+				PolicyController:           kacv1.KlusterletAddonAgentConfigSpec{Enabled: true},
```

Comment:
the `hostedAddOns` knows that only the policy addon has supported the hosted mode, so can we make the `hostedAddOns` common, and here read from it instead of hardcode which is true and which is false？

Reply:
Currently four policy add-ons support hosted mode, while only two of them will be enabled by default for cluster imported in hosted mode.